### DISCUSSION TITLE
Missing `Sidekiq::Batch::Status#success?` method

### DISCUSSION THREAD

**Discussion by rafaelmillan** [sidekiq/sidekiq#5278] [Ideas]
I found myself trying to figure out if a batch and its children batches were successful, and I realized that there is no `Sidekiq::Batch::Status#success?` method (despite there being a `Sidekiq::Batch::Status#complete?` method).

I imagine something like this would work:
```ruby
def success?
  complete? && failures == 0 && @props["cbdeath"].to_i == 0
end
```

What do you think?

**Comment by mperham** [comment]
Once a batch is successful, it is deleted from Redis so it’s pretty hard to safely provide that method. You should determine success based on side effects, ie something in your app database, set your own marker in Redis, etc.

**Reply by rafaelmillan** [reply]
I didn't think of that. I was meaning to use it in the parent batch's callback (while the batch is still in Redis) to send notifications:
```ruby
def on_complete(sidekiq_status, _options)
  if sidekiq_status.success?
    # Notify that all jobs/batches are done and successful
  else
    # Notify that all jobs/batches are done with failures
  end
end
```
I suppose that if it's too specific I could implement it myself for my use case.

**Reply by mperham** [reply]
You can definitely use a success callback in the batch or parent batch to send notifications, you just can't get a batch success reliably **outside** of the batch because of the success/deletion race condition.

**Reply by rafaelmillan** [reply]
Ok, thanks a lot for your input 🙌
